### PR TITLE
Added MonoDevelop.Debugger.PerfTests

### PR DESCRIPTION
**Pull request by jstedfast**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/605633/

### CODE REVIEW

**Comment by jstedfast** [comment]
Command failed
Stack trace:
  at MonoDevelop.UnitTesting.NUnit.NUnitAssemblyTestSuite.RunWithConsoleRunner (MonoDevelop.Core.Execution.ProcessExecutionCommand cmd, MonoDevelop.UnitTesting.UnitTest test, System.String suiteName, System.String pathName, System.String testName, MonoDevelop.UnitTesting.TestContext testContext) [0x00205] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MonoDevelop.UnitTesting.NUnit/MonoDevelop.UnitTesting.NUnit/NUnitAssemblyTestSuite.cs:537

**Comment by jstedfast** [comment]
Once again I get this type of error, but this time I have the following bits:

    <TestRunnerCommand>..\..\..\..\build\bin\mdtool.exe</TestRunnerCommand>
    <TestRunnerArgs>run-md-tests</TestRunnerArgs>

... so I have no idea how to fix this :-\

**Comment by Therzok** [comment]
@jstedfast UserInterfaceTesting is not imported by any addin, so you need to copy it locally.

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.PerfTests/MonoDevelop.Debugger.PerfTests.csproj [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    </ProjectReference>
+    <ProjectReference Include="..\..\..\core\MonoDevelop.Ide\MonoDevelop.Ide.csproj">
+      <Project>{27096E7F-C91C-4AC6-B289-6897A701DF21}</Project>
+      <Name>MonoDevelop.Ide</Name>
+      <Private>False</Private>
+    </ProjectReference>
+    <ProjectReference Include="..\..\..\..\tests\performance\MonoDevelop.PerformanceTesting\MonoDevelop.PerformanceTesting.csproj">
+      <Project>{5DC29B94-D9AF-4795-AFDD-A468ECA21A10}</Project>
+      <Name>MonoDevelop.PerformanceTesting</Name>
+      <Private>False</Private>
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.PerfTests/MonoDevelop.Debugger.PerfTests.csproj]:
```suggestion
      <Private>True</Private>
```

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.PerfTests/MonoDevelop.Debugger.PerfTests.csproj [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    </ProjectReference>
+    <ProjectReference Include="..\..\..\..\tests\ui\MonoDevelop.UserInterfaceTesting\MonoDevelop.UserInterfaceTesting.csproj">
+      <Project>{82B9A5C5-5713-49B5-93D1-BE29B00FF35A}</Project>
+      <Name>MonoDevelop.UserInterfaceTesting</Name>
+      <Private>False</Private>
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.PerfTests/MonoDevelop.Debugger.PerfTests.csproj]:
```suggestion
      <Private>True</Private>
```

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.PerfTests/MonoDevelop.Debugger.PerfTests.csproj (hunk 1 above)
Comment (jstedfast) [reply main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.PerfTests/MonoDevelop.Debugger.PerfTests.csproj]: Oh!

**Comment by jstedfast** [comment]
Hmmm, unfortunately, neither of those changes had any effect :-\

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.PerfTests/MonoDevelop.Debugger.PerfTests.csproj [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{785F775C-9CA0-4411-86E6-B89D89F3E2ED}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <RootNamespace>MonoDevelop.Debugger.PerfTests</RootNamespace>
+    <AssemblyName>MonoDevelop.Debugger.PerfTests</AssemblyName>
+    <TargetFrameworkVersion>$(MDFrameworkVersion)</TargetFrameworkVersion>
+    <TestRunnerCommand>..\..\..\..\build\bin\mdtool.exe</TestRunnerCommand>
+    <TestRunnerArgs>run-md-tests</TestRunnerArgs>
+    <OutputPath>..\..\..\build\tests</OutputPath>
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger.PerfTests/MonoDevelop.Debugger.PerfTests.csproj]: @jstedfast okay, you can set those 2 back to `<Private>false</Private>`

```suggestion
    <OutputPath>..\..\..\..\build\tests</OutputPath>
```

**Comment by sgmunn** [comment]
@jstedfast we might have to create a new release for this after all - re-deploying the perf tests didn't pick up the new test step that we added.

**Comment by Therzok** [comment]
If you modified the main release, you need to create a new release. Re-deploying does not work for that scenario.

**Comment by sgmunn** [comment]
@jstedfast the tests ran but failed right at the end.

**Comment by jstedfast** [comment]
I have no idea what the failure even is...

**Comment by sgmunn** [comment]
@jstedfast the perf test is green!

**Comment by jstedfast** [comment]
Yay!

**Pull request merged**